### PR TITLE
Log timestamps with microseconds precision

### PR DESCRIPTION
**Pull request by schnidrig**
When forwarding logs to logstash/kibana log lines get out of order since may of them have the same timestamp (which has currently only a seconds resolution).

With this patch, the precision is increased to microseconds. That way kibana can display log lines in chronological order.

### CODE REVIEW

File: tests/lib/log/owncloud.php [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU AFFERO GENERAL PUBLIC LICENSE for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public
+ * License along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+class Test_Log_Owncloud extends \PHPUnit_Framework_TestCase
```
Comment (nickvergessen) [inline tests/lib/log/owncloud.php]: Should extend `Test\TestCase` and have a namespace

File: tests/lib/log/owncloud.php [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+{
+	private $restore_logfile;
+	private $restore_logdateformat;
+
+	public function setUp() {
```
Comment (nickvergessen) [inline tests/lib/log/owncloud.php]: this method is protected not public

File: tests/lib/log/owncloud.php [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+		$restore_logfile = OC_Config::getValue("logfile");
+		$restore_logdateformat = OC_Config::getValue('logdateformat');
+		
+		OC_Config::setValue("logfile", OC_Config::getValue('datadirectory') . "/logtest");
+		OC_Log_Owncloud::init();
+	}
+	public function tearDown() {
```
Comment (nickvergessen) [inline tests/lib/log/owncloud.php]: this method is protected not public

File: tests/lib/log/owncloud.php (hunk 2 above)
Comment (nickvergessen) [inline tests/lib/log/owncloud.php]: should call `parent::setUp()` in the beginning

File: tests/lib/log/owncloud.php (hunk 3 above)
Comment (nickvergessen) [inline tests/lib/log/owncloud.php]: should call `parent::tearDown()` at the end

**Comment by karlitschek** [comment]
technically a change of the log format. But this should work. :+1:

**Comment by DeepDiver1975** [comment]
@karlitschek 8.2?

**Comment by karlitschek** [comment]
@DeepDiver1975 no preference from my side. What do you think?

**Comment by DeepDiver1975** [comment]
> What do you think?

we have more burning issue to take care of before the freeze today

**Comment by DeepDiver1975** [comment]
#20472 is merged -> closing

**Comment by PVince81** [comment]
Regression: https://github.com/owncloud/core/issues/23134

Might need to find a common ground to make it work for both situations.

**Pull request closed without merging**